### PR TITLE
Fix formatting and clippy errors

### PR DESCRIPTION
**Pull request by jcrossley3**
More work toward #1

### CODE REVIEW

**Review by bobmcwhirter — APPROVED** [review]

File: backend/api/src/system/package/mod.rs [hunk 1]
```diff
@@ -927,330 +928,283 @@ mod tests {
         Ok(())
     }
 
-    #[cfg(test)]
-    mod tests {
```
Comment (jcrossley3) [inline backend/api/src/system/package/mod.rs:931]: @bobmcwhirter This looks like more changes than it is. I just moved all the tests beneath a single `tests` module.

File: backend/common/src/purl.rs [hunk 2]
```diff
@@ -128,15 +128,18 @@ impl Debug for Purl {
...
-impl From<&&str> for Purl {
-    fn from(value: &&str) -> Self {
-        PackageUrl::from_str(value).unwrap().into()
-    }
-}
+// impl From<&&str> for Purl {
+//     fn from(value: &&str) -> Self {
+//         PackageUrl::from_str(value).unwrap().into()
+//     }
+// }
```
Comment (jcrossley3) [inline backend/common/src/purl.rs:142]: @bobmcwhirter I thought this looked weird. It's not used, but I forgot to delete it.

File: backend/common/src/purl.rs [hunk 3]
```diff
@@ -128,15 +128,18 @@ impl Debug for Purl {
 
 impl From<&str> for Purl {
     fn from(value: &str) -> Self {
-        PackageUrl::from_str(value).unwrap().into()
```
Comment (jcrossley3) [inline backend/common/src/purl.rs:131]: @bobmcwhirter We're using a clippy rule that doesn't like `unwrap`. We only allow them for tests. I don't know if deriving `Default` is The Right Thing in this case or not, but seemed harmless.

Comment (bobmcwhirter) [reply backend/common/src/purl.rs:131]: I think the answer is TryFrom instead since failure is an option.

**Pull request merged**